medicaltorch/datasets.py

```diff
@@ -87,17 +87,21 @@ def __init__(self, input_filenames, gt_filename, metadata=None, cache=True, cano
             if len(input_img.shape) > 3:
                 raise RuntimeError("4-dimensional volumes not supported.")
 
-        # we consider only one gt per patient
+        # list of GT for multiclass segmentation
+        self.gt_handle = []
+
         # Unlabeled data (inference time)
-        if self.gt_filename is None:
-            self.gt_handle = None
-        else:
-            self.gt_handle = nib.load(self.gt_filename)
+        if self.gt_filenames is not None:
+            for gt in self.gt_filenames:
+                if gt is not None:
+                    self.gt_handle.append(nib.load(gt))
+                else:
```

Comment:
Could `gt` be `None`?

Reply:
Yes if the label is empty (e.i if I don't have any cavity for a specific subject) :)